src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileCreatorTest.java

```diff
@@ -537,4 +570,80 @@ void test_createIdamRolesRequest() {
 
         assertThat(roles).isNotEmpty();
     }
+
+    @Test
+    void test_reInvite_user_successfully_with_same_idamId() throws JsonProcessingException {
+        when(userProfileRepository.findByEmail(any(String.class))).thenReturn(Optional.ofNullable(userProfile));
+        when(validationHelperService.validateReInvitedUser(any())).thenReturn(userProfile);
+        IdamFeignClient.User userResponse = new IdamFeignClient.User();
+        userResponse.setId("1234");
+        ObjectMapper mapper = new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES,
+                false);
+        String body = mapper.writeValueAsString(Set.of(userResponse));
+
+        Response userProResponse = Response.builder().request(mock(Request.class)).body(body,
+                Charset.defaultCharset()).status(200).build();
+
+        when(idamFeignClient.getUserFeed(any())).thenReturn(userProResponse);
+        userProfile.setIdamId("1234");
+        when(idamService.registerUser(any(IdamRegisterUserRequest.class))).thenReturn(idamRegistrationInfo);
+        UserProfile response = userProfileCreator.reInviteUser(userProfileCreationData);
+        assertThat(response).usingRecursiveComparison().isEqualTo(userProfile);
+        assertThat(response.getIdamRegistrationResponse()).isEqualTo(201);
+
+        InOrder inOrder = inOrder(idamService, userProfileRepository);
+        inOrder.verify(userProfileRepository, times(1)).findByEmail(any(String.class));
+        //inOrder.verify(userProfileRepository, times(1)).save(any(UserProfile.class));
+
+        verify(validationHelperService, times(1)).validateReInvitedUser(any());
+        verify(idamFeignClient, times(1)).getUserFeed(any());
+    }
+
+    @Test
+    void test_reInvite_user_failure_with_diff_idamId() throws JsonProcessingException {
```

Comment:
test method has failure , but dont see any assertions for error messages ?

Reply:
Thanks @rajkatla-hmcts changed method name as there is no failure scenario for diff idam id except updating idamid in up in case of difference ...